aptos-move/move-examples/nfts_as_accounts/sources/hero.move

```diff
@@ -0,0 +1,287 @@
+/// This demonstrates how to make a composable NFT on Aptos. We take a Hero that can have optional
+/// weapons, armor, and a shield. Each of these can optionally have a gem. These can be added or
+/// removed and seamlessly treated as Tokens or their richer type.
+///
+/// todo(@davidiw):
+/// * complete armor and shield
+/// * add accessors to everything
+/// * make it possible to get a reference to the various ref types so that the items can remain
+///   as is without requiring decomposing
+/// * better error handling around if an item is already equipped
+/// * add the ability to remove weapons and gems
+module nfts_as_accounts::composable_example {
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{Self, String};
+
+    use nfts_as_accounts::token;
+
+    const ENOT_A_HERO: u64 = 1;
+    const ENOT_A_WEAPON: u64 = 2;
+    const ENOT_A_GEM: u64 = 3;
+
+    struct OnChainConfig has key {
+        collection: String,
+        mutability_config: token::MutabilityConfig,
+        royalty: token::Royalty,
+    }
+
+    struct Hero has key {
```

Comment:
do we need at this point to call it token or have a token_ref? we can build a market for any struct data?

Reply:
I mean this could be made an object, but the point of it being a token is the common data in the token itself. If we like this and we want further generalization, I'd recommend

Object<Token<Data>>